src/material-experimental/mdc-form-field/form-field.ts

```diff
@@ -668,9 +661,8 @@ export class MatFormField
       ((iconPrefixContainer ? iconPrefixContainerWidth - WRAPPER_HORIZONTAL_PADDING : 0) +
         textPrefixContainerWidth);
 
-    // Update the transform the floating label to account for the prefix container. Note
-    // that we do not want to overwrite the default transform for docked floating labels.
-    floatingLabel.style.transform = `${FLOATING_LABEL_DEFAULT_DOCKED_TRANSFORM} translateX(${labelHorizontalOffset}px)`;
+    // Update the transform x of the floating label to account for the prefix container.
+    floatingLabel.style.transform = `var(--mat-mdc-form-field-label-transform, translateX(${labelHorizontalOffset}px)`;
```

Comment:
This seems a little weird to me. IMO the `var(...)` should be in the structural styles and the runtime only sets the CSS variable value.

Reply:
Yes ideally if I had control over all of the styles I would do it that way. However, MDC's mixins add `transform` properties in a number of different places that wind up conflicting with it if I try to add it in the CSS. Instead this keeps the CSS basically how it is today, but changes the override via `!important` to an override via defining a CSS var